technical-fundamentals/coding/codeReview/index.mjs

```diff
@@ -1,50 +1,50 @@
 // non-declarative naming
-function mapArray(array) {
-  return array.map((el) => !!el);
+function mapToBooleans(array) {
+  return array.map(Boolean);
 }
 
 class Player {}
 let player;
 
 // declarative naming mismatch
-function getPlayer() {
+function getOrCreatePlayer() {
   if (!player) {
     player = new Player();
   }
   return player;
 }
 
 // use named parameters
-function evaluateChallenge(challenge, result, candidate, difficulty) {
+function evaluateChallenge({ challenge, result, candidate, difficulty }) {
   if (challenge === result) {
     return `${candidate} has successfully completed the ${difficulty} challenge`;
   }
   return `${candidate}'s submission is incorrect`;
 }
 
 // Oversplitting functions - "The rule of 3"
+function wordCounter(string) {
+  const words = string.split(" ");
 
-function splitWords(str) {
-  return str.split(" ");
-}
-function wordMapper(words) {
-  hash = {};
-  return words.map((word) => (hash[word] = hash[word] ? 0 : (hash[word] += 1)));
-}
+  const wordCount = words.reduce((hash, word) => {
+    hash[word] ??= 0;
 
-function wordCounter(string) {
-  return wordMapper(splitWords(string));
+    hash[word]++;
+
+    return hash;
+  }, {});
+
+  return wordCount;
 }
 
 // Avoid sideEffects
 
 const tracker = {};
 
 function countViews(key) {
-  if (!tracker[key]) {
-    tracker[key] = 0;
-  }
-  return ++tracker[key];
+  const views = tracker[key] || 0;
+
+  return views + 1;
```

Comment:
This function has a sideEffect - as you countthem, you modify them. Functions should be named as causing an effect, or avoid them altogether